src/main/java/com/cleanroommc/groovyscript/compat/mods/advancedrocketry/BaseRegistry.java

```diff
@@ -0,0 +1,195 @@
+package com.cleanroommc.groovyscript.compat.mods.advancedrocketry;
+
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.ingredient.OreDictIngredient;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import org.jetbrains.annotations.Nullable;
+import zmaster587.libVulpes.interfaces.IRecipe;
+import zmaster587.libVulpes.recipe.RecipesMachine;
+import zmaster587.libVulpes.tile.multiblock.TileMultiblockMachine;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public abstract class BaseRegistry extends VirtualizedRegistry<IRecipe> {
+
+    protected abstract Class<?> getMachineClass();
+
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    protected List<IRecipe> getRecipeList() {
+        Class<?> clazz = getMachineClass();
+        RecipesMachine registry = RecipesMachine.getInstance();
+        List<IRecipe> recipes = registry.getRecipes(clazz);
+        if (recipes == null) {
+            recipes = new LinkedList<>();
+            // NOTE: this cast is completely invalid but Advanced Rocketry's code is pepega
+            // @see zmaster587.libVulpes.recipe.RecipesMachine#recipeList
+            // @see zmaster587.advancedRocketry.block.BlockSmallPlatePress#getRecipe

```

Comment:
`@see` wont actually work unless its a javadoc comment (starting with `/**`) and those need to be on methods/fields/classes. additionally, could you explain the issue better than "pepega" - perhaps "uses BlockSmallPlatePress as a generic class in a location that should ..."

Reply:
Honestly thinking about it, this has no place in a docstring. The docstring's purpose is to show how the method is used, not why there's a generic modifier missing. I'll remove it.